Library/Homebrew/linkage_checker.rb

```diff
@@ -40,18 +41,14 @@ def check_dylibs
       end
     end
 
-    begin
-      f = Formulary.from_rack(keg.rack)
-      f.build = Tab.for_keg(keg)
+    if formula
       filter_out = proc do |dep|
-        dep.build? || (dep.optional? && !dep.option_names.any? { |n| f.build.with?(n) })
+        dep.build? || (dep.optional? && !dep.option_names.any? { |n| formula.build.with?(n) })
       end
-      declared_deps = f.deps.reject { |dep| filter_out.call(dep) }.map(&:name) +
-                      f.requirements.reject { |req| filter_out.call(req) }.map(&:default_formula).compact
+      declared_deps = formula.deps.reject { |dep| filter_out.call(dep) }.map(&:name) +
+                      formula.requirements.reject { |req| filter_out.call(req) }.map(&:default_formula).compact
       @undeclared_deps = @brewed_dylibs.keys - declared_deps.map { |dep| dep.split("/").last }
-      @undeclared_deps -= [f.name]
-    rescue FormulaUnavailableError
-      opoo "Formula unavailable: #{keg.name}"
+      @undeclared_deps -= [formula.name]
```

Comment:
Since you have to make changes to this entire code block: I think it would be better to use an early exit with `return unless formula` or better yet relocate the calculations done here into a separate (private) method, e.g.:

```ruby
@undeclared_deps = find_undeclared_deps if formula
```